metricbeat/docs/modules/azure.asciidoc

```diff
@@ -10,6 +10,76 @@ beta[]
 
 This is the azure module.
 
+The Azure Monitor feature collects and aggregates logs and metrics from a variety of sources into a common data platform where it can be used for analysis, visualization, and alerting.
+
+
+The azure monitor metrics are numerical values that describe some aspect of a system at a particular point in time. They are collected at regular intervals and are identified with a timestamp, a name, a value, and one or more defining labels.
+This is the list of supported metrics.
+
+The azure module will periodically retrieve the azure monitor metrics using the Azure REST APIs as MetricList.
+Additional azure API calls will be executed in order to retrieve information regarding the resources targeted by the user.
+
+The azure module mericsets are `monitor`,  `compute_vm` and `compute_vm_scaleset`
+
+[float]
+=== Module-specific configuration notes
+
+All the tasks executed against the Azure Monitor  REST API will use the Azure Resource Manager authentication model.
+Therefore, all requests must be authenticated with Azure Active Directory (Azure AD).
+One approach to authenticate the client application is to create an Azure AD service principal and retrieve the authentication (JWT) token.
+For a more detailed walk-through, have a look at using Azure PowerShell to create a service principal to access resources. It is also possible to create a service principal via the Azure portal.
+
+Required credentials for the `azure` module:
+
+`client_id` - Unique identifier for the application (also known as Application Id)
+
+`client_secret` - The client/application secret/key
+
+`subscription_id` - Unique identifier for the azure subscription
+
+`tenant_id` - The unique identifier of the Azure Active Directory instance
+
+Users can use the azure credentials keys if configured `AZURE_CLIENT_ID`, `AZURE_CLIENT_SECRET`, `AZURE_TENANT_ID`, `AZURE_SUBSCRIPTION_ID`
```

Comment:
I have cleaned up the documentation and touched on your findings as well. At the moment this is the only place where it will be used. Do you have any recommendations where the information regarding credentials should be moved?

Reply:
We have some shared doc files in `libbeat/docs/shared-*`. We can wait to move this content to a shared file when we need it in any case.

Reply:
sounds good